src/views/Messages/MessageList.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright © 2021 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useEffect, useContext } from 'react';
+import { useHistory } from 'react-router';
+import { TablePagination, makeStyles } from '@material-ui/core';
+import dayjs from 'dayjs';
+import { useTranslation } from 'react-i18next';
+import { IMessage, IDataTableRecord, IHistory } from '../../interfaces';
+import { DataTable } from '../../components/DataTable/DataTable';
+import { HashPopover } from '../../components/HashPopover';
+import { ApplicationContext } from '../../contexts/ApplicationContext';
+import { NamespaceContext } from '../../contexts/NamespaceContext';
+
+interface Props {
+  setViewMessage: React.Dispatch<React.SetStateAction<IMessage | undefined>>;
+}
+
+const PAGE_LIMITS = [10, 25];
+
+export const MessageList: React.FC<Props> = ({ setViewMessage }) => {
+  const history = useHistory<IHistory>();
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { selectedNamespace } = useContext(NamespaceContext);
+  const { createdFilter, lastEvent } = useContext(ApplicationContext);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(PAGE_LIMITS[0]);
+  const [messages, setMessages] = useState<IMessage[]>([]);
+
+  const columnHeaders = [
+    t('author'),
+    t('type'),
+    t('tag'),
+    t('transactionType'),
+    t('dataHash'),
+    t('createdOn'),
+  ];
+
+  useEffect(() => {
+    let createdFilterString = `&created=>=${dayjs()
+      .subtract(24, 'hours')
+      .unix()}`;
+    if (createdFilter === '30days') {
+      createdFilterString = `&created=>=${dayjs().subtract(30, 'days').unix()}`;
+    }
+    if (createdFilter === '7days') {
+      createdFilterString = `&created=>=${dayjs().subtract(7, 'days').unix()}`;
+    }
+
+    fetch(
```

Comment:
I separated the queries between list and timeline view in preparation for moving the timeline to use https://react-query.tanstack.com/guides/infinite-queries for infinite scroll